### ISSUE TITLE
Image preview doesn't fix relative paths, leading to 404s

### ISSUE THREAD

**Issue by itamarst** [CloudCannon/pagefind#78]
I have an image that uses a relative path. When it gets previewed in search from a different directory, it is still a relative path so the result is a 404.

E.g. `/foo/` has `<img src="foo.png">`.

When visiting `/bar/` and doing a search, it ought to load `/foo/foo.png`, but instead sticks to loading just `foo.png`, so it loads `/bar/foo.png` which then fails (or might load wrong image...).

**Comment by bglw** [comment]
Ah, relative paths 😅

The tricky thing here is that this is just [Pagefind metadata](https://pagefind.app/docs/metadata/), which doesn't have any processing on it, and I'm a little hesitant to change that.

One path would be to only add url processing to the automatic `image` metadata, but then people would see a different result if they manually tagged an image with `data-pagefind-meta="image[src]"`. 

I could make Pagefind perform url processing automatically if it sees metadata named `image`, or if it grabs metadata from a `src` or `href` attribute, but I fear doing this in all cases as it might _not_ be what someone expects for their workflow.

Maybe the best option is to handle this in the Pagefind UI specifically. I'll look into handling it there, with the small downside that someone building their own Pagefind widget will also have to implement similar behavior if they encounter relative urls.

**Comment by itamarst** [comment]
I fixed it personally by just changing those paths to be absolute.

**Comment by tabris17** [comment]
Is it possible to provide a hook to handle the search results in PagefindUI constructor arguments?

```
<script>
    window.addEventListener('DOMContentLoaded', (event) => {
        new PagefindUI({ 
            element: "#search",
            processResult: function (result) {
                result.meta.image = convertRelativePathToAbsolutePath(result.meta.image)
                return result
            }
        });
    });
</script>
```

**Comment by bglw** [comment]
Oh, that is a fantastic idea. I'll make that happen.

**Comment by bglw** [comment]
Hello! A `processResult` hook has been implemented in the [Pagefind v0.9.0](https://github.com/CloudCannon/pagefind/releases/tag/v0.9.0) release 🎉
